code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,477 @@
+GLOBAL_LIST_INIT(waves, list("small wave" = list(/obj/structure/crystal_portal/small=4, /obj/structure/crystal_portal/medium=1),
```

Comment:
I'm not convinced these actually need to be globals but this name is not good. Should at the very least be renamed to ``crystal_invasion_waves"

Reply:
done